packages/@aws-cdk/aws-ecs/lib/firlens-log-router.ts

```diff
@@ -0,0 +1,204 @@
+ */
+export enum FirelensConfigFileType {
+  /**
+   * s3
+   */
+  S3 = 's3',
+
+  /**
+   * fluentd
+   */
+  FILE = 'file',
+}
+
+/**
+ * The options for firelens log router
+ * https://docs.aws.amazon.com/AmazonECS/latest/developerguide/using_firelens.html#firelens-taskdef-customconfig
+ */
+export interface FirelensOptions {
+  /**
+   * By default, Amazon ECS adds additional fields in your log entries that help identify the source of the logs.
+   * You can disable this action by setting enable-ecs-log-metadata to false.
+   * @default - true
+   */
+  readonly enableECSLogMetadata?: boolean;
+
+  /**
+   * Custom configuration file, s3 or file
+   * @default - determined by checking configFileValue with S3 ARN.
+   */
+  readonly configFileType?: FirelensConfigFileType;
+
+  /**
+   * Custom configuration file, S3 ARN or a file path
+   */
+  readonly configFileValue: string;
+}
+
+/**
+ * Firelens Configuration
+ * https://docs.aws.amazon.com/AmazonECS/latest/developerguide/using_firelens.html#firelens-taskdef
+ */
+export interface FirelensConfig {
+
+  /**
+   * The log router to use
+   * @default - fluentbit
+   */
+  readonly type: FirelensLogRouterType;
+
+  /**
+   * Firelens options
+   * @default - no additional options
+   */
+  readonly options?: FirelensOptions;
+}
+
+/**
+ * The properties in a firelens log router.
+ */
+export interface FirelensLogRouterProps extends ContainerDefinitionProps {
+  /**
+   * Firelens configuration
+   */
+  readonly firelensConfig: FirelensConfig;
+}
+
+/**
+ * The options for creating a firelens log router.
+ */
+export interface FirelensLogRouterDefinitionOptions extends ContainerDefinitionOptions {
+  /**
+   * Firelens configuration
+   */
+  readonly firelensConfig: FirelensConfig;
+}
+
+/**
+ * Render to CfnTaskDefinition.FirelensConfigurationProperty from FirelensConfig
+ */
+function renderFirelensConfig(firelensConfig: FirelensConfig): CfnTaskDefinition.FirelensConfigurationProperty {
+  if (!firelensConfig.options) {
+    return { type: firelensConfig.type };
+  } else {
+    const enableECSLogMetadata = firelensConfig.options.enableECSLogMetadata || firelensConfig.options.enableECSLogMetadata === undefined ? 'true' : 'false';
+    const configFileType = firelensConfig.options.configFileType || /arn:aws[a-zA-Z-]*:s3:::.+/.test(firelensConfig.options.configFileValue)
+      ? FirelensConfigFileType.S3 : FirelensConfigFileType.FILE;
+    return {
+      type: firelensConfig.type,
+      options: {
+        'enable-ecs-log-metadata': enableECSLogMetadata,
+        'config-file-type': configFileType,
+        'config-file-value': firelensConfig.options.configFileValue
+      }
+    };
+  }
+}
+
+/**
+ * SSM parameters for latest fluent bit docker image in ECR
+ * https://github.com/aws/aws-for-fluent-bit#using-ssm-to-find-available-versions
+ */
+const latestFluentBitImage = '/aws/service/aws-for-fluent-bit/2.1.0';
```

Comment:
Hardcoding an image version isn't very maintainable; we don't want this to get out of date and we don't want to have to keep updating it all the time.

You should probably use the latest tag. Another good option is to always have users specify a version number. Allow them to input "2.1.0" and then construct the SSM parameter name from that.

Reply:
There's no easy way to fetch the latest (rather than 'latest') version(e.g. 2.1.0) by running following API/CLI in CDK:

``` 
aws ssm get-parameters-by-path --path /aws/service/aws-for-fluent-bit/ --query 'Parameters[*].Name'
```

so I added the optional image tag parameter in this function, the default value is 'latest' if it's not defined.